src/tacticals.fun

```diff
@@ -51,6 +105,13 @@ struct
   fun THEN (tac1, tac2) =
     THEN_LAZY (tac1, fn () => tac2)
 
+  (* TODO: Should ORELSE_LAZY have some idea of
+   * how to keep track of failing tactics to present
+   * a summary to the user?
+   *
+   * This wouldn't be hard to implement and potentially makes
+   * debugging much easier.
```

Comment:
that would be pretty interesting! I would probably accept a pull request to that effect.